src/main/scala/io/lenses/connect/secrets/providers/VaultSecretProvider.scala

```diff
@@ -133,8 +132,15 @@ class VaultSecretProvider() extends ConfigProvider with VaultHelper {
           )
         }
 
-        val ttl = Option(vaultClient.get.logical().read(path).getLeaseDuration) match {
-          case Some(duration) => Some(now.plusSeconds(duration))
+        val ttl = Option(response.getLeaseDuration) match {
+          case Some(duration) =>
+            if (duration != 0) {
+              Some(now.plusSeconds(duration))
+            } else if (settings.secretTtlDefault > 0) {
+              Some(now.plusSeconds(settings.secretTtlDefault))
+            } else {
+              None
```

Comment:
Shouldn't this be `Some(now)` given that previously with a duration, we returned `now.plusSeconds(duration)`?